core/src/main/java/org/apache/accumulo/core/iterators/IteratorEnvironment.java

```diff
@@ -140,7 +140,8 @@ default SamplerConfiguration getSamplerConfiguration() {
   }
 
   /**
-   * True if compaction was user initiated.
+   * True if compaction was user initiated. Will throw IllegalStateException if
+   * {@link #getIteratorScope()} != {@link IteratorScope#majc}.
```

Comment:
```suggestion
   * @return True if compaction was user initiated, false otherwise.
   * @throws IllegalStateException if {@link #getIteratorScope()} != {@link IteratorScope#majc}
```